pkg/controller/operands/ssp_test.go

```diff
@@ -70,822 +79,549 @@ var _ = Describe("SSP Operands", func() {
 		})
 
-		// TODO: add tests to ensure that HCO properly propagates NodePlacement from its CR
-
-		// TODO: temporary avoid checking conditions on KubevirtCommonTemplatesBundle because it's currently
-		// broken on k8s. Revert this when we will be able to fix it
-		/*
-			It("should handle conditions", func() {
-				expectedResource := newKubeVirtCommonTemplateBundleForCR(hco, OpenshiftNamespace)
-				expectedResource.ObjectMeta.SelfLink = fmt.Sprintf("/apis/v1/namespaces/%s/dummies/%s", expectedResource.Namespace, expectedResource.Name)
-				expectedResource.Status.Conditions = []conditionsv1.Condition{
-					conditionsv1.Condition{
-						Type:    conditionsv1.ConditionAvailable,
-						Status:  corev1.ConditionFalse,
-						Reason:  "Foo",
-						Message: "Bar",
-					},
-					conditionsv1.Condition{
-						Type:    conditionsv1.ConditionProgressing,
-						Status:  corev1.ConditionTrue,
-						Reason:  "Foo",
-						Message: "Bar",
-					},
-					conditionsv1.Condition{
-						Type:    conditionsv1.ConditionDegraded,
-						Status:  corev1.ConditionTrue,
-						Reason:  "Foo",
-						Message: "Bar",
-					},
-				}
-				cl := initClient([]runtime.Object{hco, expectedResource})
-				r := initReconciler(cl)
-				Expect(r.ensureKubeVirtCommonTemplateBundle(req)).To(BeNil())
-
-				// Check HCO's status
-				Expect(hco.Status.RelatedObjects).To(Not(BeNil()))
-				objectRef, err := reference.GetReference(r.scheme, expectedResource)
-				Expect(err).To(BeNil())
-				// ObjectReference should have been added
-				Expect(hco.Status.RelatedObjects).To(ContainElement(*objectRef))
-				// Check conditions
-				Expect(req.Conditions[]).To(ContainElement(testlib.RepresentCondition(conditionsv1.Condition{
-					Type:    conditionsv1.ConditionAvailable,
-					Status:  corev1.ConditionFalse,
-					Reason:  "KubevirtCommonTemplatesBundleNotAvailable",
-					Message: "KubevirtCommonTemplatesBundle is not available: Bar",
-				})))
-				Expect(req.Conditions[]).To(ContainElement(testlib.RepresentCondition(conditionsv1.Condition{
-					Type:    conditionsv1.ConditionProgressing,
-					Status:  corev1.ConditionTrue,
-					Reason:  "KubevirtCommonTemplatesBundleProgressing",
-					Message: "KubevirtCommonTemplatesBundle is progressing: Bar",
-				})))
-				Expect(req.Conditions[]).To(ContainElement(testlib.RepresentCondition(conditionsv1.Condition{
-					Type:    conditionsv1.ConditionUpgradeable,
-					Status:  corev1.ConditionFalse,
-					Reason:  "KubevirtCommonTemplatesBundleProgressing",
-					Message: "KubevirtCommonTemplatesBundle is progressing: Bar",
-				})))
-				Expect(req.Conditions[]).To(ContainElement(testlib.RepresentCondition(conditionsv1.Condition{
-					Type:    conditionsv1.ConditionDegraded,
-					Status:  corev1.ConditionTrue,
-					Reason:  "KubevirtCommonTemplatesBundleDegraded",
-					Message: "KubevirtCommonTemplatesBundle is degraded: Bar",
-				})))
+		Context("NodeLabeller node placement", func() {
```

Comment:
This is a bit misleading to me. This Context and the one in line 248 uses the old names and I find it confusing. But I undersand the logic. Your call...

Also, since workloads and infra are not depeneded in each other, do we reaally need this split in the contexts. Maybe we do it inside the same one?

Reply:
The use of `NodeLabeller` and `TemplateValidator` don't refer to the old CRDs, but rather to the components (which still exist) and their corresponding fields within `SSP.Spec`.

About merging both under the same context... I was considering this too while implementing this, but thought that it would be good to test for them being and staying independent when the code change. But rethinking this now, I think it'd be better to shrink the test code in ~140 lines vs. future-proofing from unlikely bugs. So, will merge the contexts.

Reply:
Done.